app/utility/entity_merger.rb

```diff
@@ -1,192 +1,348 @@
+      merge_os_donations!
+      merge_ny_donations!
+      transfer_ny_filer_entity!
+      replace_os_match_cmte_id!
+      set_merged_id_and_delete
     end
-
-    e1
   end
 
-  def self.merge_attribute(key, old, nu)
-    return nu if old.nil?
-    return old if nu.nil?
-    return old if old == nu
+  # trial run
+  def report
+    merge
+    cp = ColorPrinter
+    puts "Merging #{cp.colorize(source.name, :red)} (#{cp.colorize(source.id, :bg_red) }) into #{cp.colorize(dest.name, :red)} (#{cp.colorize(dest.id, :bg_red)})"
 
-    append_with_new_line_fields = ['summary', 'notes']
-    append_with_space_fields = ['name_suffix', 'name_prefix']
-    date_fields = ['start_date', 'end_date']
-    name_fields = ['name_first', 'name_middle']
-    number_fields = ['employees', 'revenue', 'endowment', 'net_worth']
+    unless @extensions.length.zero?
+      puts cp.cyan("Adding or updating ") + cp.red(@extensions.count.to_s) + cp.cyan(" extensions")
+    end
 
-    return old + "\n\n" + nu if append_with_new_line_fields.include?(key)
-    return old + " " + nu if append_with_space_fields.include?(key)
+    unless @contact_info.length.zero?
+      puts cp.cyan("Transferring ") + cp.red(@contact_info.count.to_s) + cp.cyan(" contact info models")
+    end
 
-    if date_fields.include?(key)
-      old_date = Date.new(old)
-      nu_date = Date.new(nu)
+    unless @lists.length.zero?
+      puts cp.cyan("Putting the merged entity on ") + cp.red(@lists.count.to_s) + cp.cyan(" new lists")
+    end
 
-      return old_date.format if old_date.how_specific > nu_date.how_specific
-      return nu_date.format if old_date_how_specific < nu_date.how_specific
+    unless @relationships.length.zero?
+      puts cp.cyan("Transfering ") + cp.red(@relationships.count.to_s) + cp.cyan(" Relationships")
+    end
 
-      return (Date.less_or_equal(old_date, nu_date) ? old_date : nu_date) if old_date.how_specific == Date::YEAR_SPECIFIC
-      return old_date
+    unless @potential_duplicate_relationships.length.zero?
+      puts cp.red("NOTICE: ") + cp.cyan("found ") + cp.blue(@potential_duplicate_relationships.count.to_s) + cp.cyan(' potential duplicate relationships')
     end
 
-    return ((old.length >= nu.length or old.length != 1) ? old : nu) if name_fields.include?(key)
-    return (old == 0 ? nu : old) if number_fields.include?(key)
+    unless @aliases.length.zero?
+      puts cp.cyan("adding ") + cp.red(@aliases.length.to_s) + cp.cyan(" aliases: ") + cp.blue(@aliases.map(&:name).join(', '))
+    end
+
+    unless @document_ids.length.zero?
+      puts cp.cyan("Transfering ") + cp.red(@document_ids.count.to_s) + cp.cyan(" documents")
+    end
 
-    old
+    unless @tag_ids.length.zero?
+      puts cp.cyan("Transfering ") + cp.red(@tag_ids.count.to_s) + cp.cyan(" tags: ") + cp.blue(@tag_ids.map { |t| Tag.find(t).name }.join(', '))
+    end
   end
 
-  def self.merge_all(e1, e2)
-    ActiveRecord::Base.transaction do
-      e1 = merge_basic(e1, e2)
+  def merge
```

Comment:
ditto, above comment!

note: since i am not going to be maintaining this code, i started skimming about halfway through the various merge methods. they all seem very clean and easy to reason about. it seems as though you are using some shared conventions between them, which makes it easier to jump from merge function to merge function to see what's going on -- or to add new ones. 

when and if this code base grows, i'd consider whether there are further conventions -- or abstractions -- available to concretise the pattern of merging to make it easy to declaratively add new things to be merged. otherwise: great work here!

Reply:
that's interesting. it's true that if we ever change how anything related to entities work -- even something like adding tags -- EntityMerger would have to be modified.